### PR TITLE
Further command implementation/syntax

### PR DESCRIPTION
**Pull request by windy1**
Added a bit more syntax for the commands. Syntax tested and working for Spout.

Signed-off-by: WalkerCrouse wjcrouse913@aim.com

### CODE REVIEW

**Comment by zml2008** [comment]
These command implementations would be terrible to maintain and use ints where string values should be used instead. Also, there are methods in CommandContext that handle int checking for you and for errors you throw a CommandException.

**Comment by windy1** [comment]
@zml2008 Oh, I didn't know we wanted string values for modes and what not I can fix the other things though if you think it's worth it.

**Comment by windy1** [comment]
Man, this is where I wish I had String switch statements now :(

**Comment by windy1** [comment]
Okay, cleaned it up quite a bit

**Comment by windy1** [comment]
Okay added String support

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java [hunk 1]
```diff
@@ -25,38 +25,160 @@
...
 	@CommandPermissions("vanilla.command.xp")
 	public void xp(CommandContext args, CommandSource source) throws CommandException {
-		source.sendMessage("Xp worked.");
-		//TODO implement xp.
+		// If source is player
+		if (args.length() == 1) {
+			if (source instanceof Player) {
+				Player sender = (Player) source;
+				int amount = args.getInteger(0);
+				source.sendMessage("You have been given " + amount + " xp.");
```
Comment (Zidane) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:122]: No reason to make a casted Player object and not even use it to send the message eh?

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java [hunk 2]
```diff
@@ -25,38 +25,160 @@
...
+		
+		// If source is not player
+		if (args.length() == 2) {
+			Player player = Spout.getGame().getPlayer(args.getString(0), true);
+			if (player != null) {
+				int amount = args.getInteger(1);
+				player.sendMessage("You have been given " + amount + " xp.");
+				// TODO: Give player 'amount' of xp.
+			} else {
+				throw new CommandException(args.getString(0) + " is not online.");
```
Comment (Zidane) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:137]: Anyone else hope to add xp to offline players? How about you @zml2008?

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java (hunk 1 above)
Comment (windy1) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:122]: That's for the to-come implementation of switching the gamemode which will require a player instance. (Doesn't really _need_ to be there atm)

**Comment by Zidane** [comment]
Only reason I have not pulled this @Windwaker is that it is not neccessary to have some int checking method. Please observe these helper methods in CommandContext: http://jd.spout.org/latest/org/spout/api/command/CommandContext.html

**Comment by windy1** [comment]
@NinjaZidane It's to see if it's a String or an int. I know I could just use args.getInteger(foo); but I'm checking so someone could use /gm creative or something similar. I still think it's justified.

**Comment by Zidane** [comment]
True as well @WalkerCrouse, perhaps you could instead do a 

try{ getInteger(blah, blah)} catch

Use our stuff and not the scanner stuff :p

**Comment by windy1** [comment]
But that's yucky :(, there should be an isInteger() in SpoutAPI then...

**Comment by Zidane** [comment]
Yeah...now that I think about it.

Rebase it and I will pull it. We can modify it later if need be (besides the implementation).

**Comment by windy1** [comment]
@NinjaZidane It's squashed... :p

**Comment by Zidane** [comment]
No...I mean rebase it with the current head. I can't merge it automatically :p

**Comment by windy1** [comment]
See https://github.com/SpoutDev/Vanilla/pull/33

**Pull request closed without merging**